tests/test_deregister_task_definition.py

```diff
@@ -0,0 +1,232 @@
+                cluster="",
+                production_task_definition_output=None,
+                preflight_task_definition_output=None,
+                local_task_definition_output=None,
+            )
+            ecs_describe_services_patch.assert_called()
+            ecs_deregister_patch.assert_not_called()
+
+    def test_deregister_task_definition_failed_pipeline(self):
+        local_exec_task_definition_1 = Mock(return_value="local_exec_arn_1")
+        with (
+            self.subTest("Failed initial deployment"),
+            self.assertRaises(SystemExit),
+            patch.object(
+                self.ecs_client,
+                "describe_services",
+                return_value=self.describe_service_return_value("PRIMARY", arn=""),
+            ) as ecs_describe_services_patch,
+            patch.object(self.ecs_client, "deregister_task_definition") as ecs_deregister_patch,
+        ):
+            deregister_task_definition(
+                ecs_client=self.ecs_client,
+                service="",
+                cluster="",
+                local_task_definition_output=CreateTaskDefinitionOutput(
+                    previous_task_definition_arn="",
+                    latest_task_definition_arn=local_exec_task_definition_1,
+                ),
+                production_task_definition_output=None,
+                preflight_task_definition_output=None,
+            )
+            ecs_describe_services_patch.assert_called()
+            ecs_deregister_patch.assert_called_once_with(
+                local_exec_task_definition_1,
+            )
+
+        with (
+            self.subTest("Failed deployment"),
+            self.assertRaises(SystemExit),
+            patch.object(
+                self.ecs_client,
+                "describe_services",
+                return_value=self.describe_service_return_value("PRIMARY", arn=""),
+            ) as ecs_describe_services_patch,
+            patch.object(self.ecs_client, "deregister_task_definition") as ecs_deregister_patch,
+        ):
+            local_exec_task_definition_2 = Mock(return_value="local_exec_arn_2")
+            deregister_task_definition(
+                ecs_client=self.ecs_client,
+                service="",
+                cluster="",
+                local_task_definition_output=CreateTaskDefinitionOutput(
+                    previous_task_definition_arn=local_exec_task_definition_1,
+                    latest_task_definition_arn=local_exec_task_definition_2,
+                ),
+                production_task_definition_output=None,
+                preflight_task_definition_output=None,
+            )
+            ecs_describe_services_patch.assert_called()
+            ecs_deregister_patch.assert_called_once_with(
+                local_exec_task_definition_2,
+            )
+
+    def test_deregister_task_definition_successful_initial_deploy(self):
+        local_task_definition_1 = Mock(return_value="local_exec_arn_1")
```

Comment:
For my understanding, do you really need to define a return value here? I thought that 
```suggestion
        local_task_definition_1 = Mock()
```
would be enough here, as `local_task_definition_1` does not get called as a function.